functions/uploads.py

```diff
@@ -117,9 +117,11 @@ def ingest_upload(event: dict, context: BackgroundContext):
                 trial_id, job.assay_type, "Assay Metadata", xlsx_blob, session=session
             )
 
-        # Save the upload success and trigger email alert
-        job.ingestion_success(session=session, send_email=True)
-        session.commit()
+        # Update the job metadata to include artifacts
+        job.assay_patch = metadata_with_urls
+
+        # Save the upload success and trigger email alert if transaction succeeds
+        job.ingestion_success(session=session, send_email=True, commit=True)
```

Comment:
Now, this call will try to commit the transaction *before* sending a "successful upload" email.